client/core/core.go

```diff
@@ -3283,8 +3296,10 @@ func (c *Core) Logout() error {
 	// Lock wallets
 	for _, w := range c.xcWallets() {
 		if w.connected() {
-			if err := w.Lock(); err != nil {
-				return err
```

Comment:
This was a major contributor to hard kill user behavior

Reply:
Any ideas what the hold up was? I'm in favor of this secondary timeout, but I do wonder what the wallet thinks it's waiting on.

Reply:
Unclear, but personally I have had bitcoin hang on RPCs when there is peer trouble, and presumably some broadly-scoped internal mutex (in bitciond) is locked for a long time that holds up all RPCs.